onnxruntime/core/graph/graph.cc

```diff
@@ -4275,6 +4285,12 @@ ONNX_NAMESPACE::GraphProto Graph::ToGraphProtoWithExternalInitializers(
     ORT_THROW("Failed to flush file with external initializers: ", modified_external_file_path);
   }
 
+  // Delete if the external data file is empty
+  if (external_empty_pos == external_stream.tellp()) {
```

Comment:
Related to comment https://github.com/microsoft/onnxruntime/pull/23498/files#r1933061801

Could we remove this code if we call `Model::Save()` (instead of `Model::SaveWithExternalInitializers()`) when we know the external file will be empty?

Reply:
1. It's a bug in the SaveWithExternalInitializers() as it creates the external file first regardless whether really has any initializers to write into the file. So, it creates an empty file if there's no initializers exceed the threshold. There's python test code hides this bug.
2. We shouldn't call Model::Save() since it do nothing to the existing external initializer which end up with some initializer still point to the old external file.

Reply:
I see. thanks